libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -455,12 +456,29 @@ Compass::init()
         // detect available backends. Only called once
         _detect_backends();
     }
-    if (_compass_count != 0) {
-        // get initial health status
-        hal.scheduler->delay(100);
+    initialise_health_status();
+    return true;
+}
+
+void Compass::initialise_health_status(void)
+{
+    if (_compass_count == 0) {
+        return;
+    }
+
+    // get initial health status
+    for (uint8_t attempts=0; attempts<10; attempts++) {
         read();
+        for (uint8_t i=0; i < COMPASS_MAX_INSTANCES; i++) {
+            if (!_state[i].healthy) {
+                hal.scheduler->delay(10);
+                continue;
+            }
+        }
+        gcs().send_text(MAV_SEVERITY_INFO, "All compasses healthy after %ums",
```

Comment:
Do we need this message that all is fine?  I'd say we should restrict it to messages when things are not going well.